src/Acuminator/Acuminator.Analyzers/Refactorings/EventHandlerSignature/EventHandlerSignatureRefactoring.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Acuminator.Utilities.Roslyn.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Acuminator.Analyzers.Refactorings.EventHandlerSignature
+{
+	[ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+	public class EventHandlerSignatureRefactoring : CodeRefactoringProvider
+	{
+		private const string ArgsParameterName = "e";
+		private const string EventHandlerMethodName = "_";
+
+		public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+			var pxContext = new PXContext(semanticModel.Compilation);
+
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+			var methodNode = root?.FindNode(context.Span)?.GetDeclaringMethodNode();
+
+			if (methodNode != null)
+			{
+				IMethodSymbol methodSymbol = semanticModel.GetDeclaredSymbol(methodNode, context.CancellationToken);
+
+				if (IsSuitableForRefactoring(methodSymbol, pxContext))
+				{
+					var eventHandlerInfo = methodSymbol.GetEventHandlerInfo(pxContext);
+
+					if (eventHandlerInfo.eventSignatureType == EventHandlerSignatureType.Default
```

Comment:
@dropsonic This condition is quite heavy to read even in IDE. you should put part of it in a variable like `key` or maybe put the whole check in a separate method

Reply:
@SENya1990 swapped conditions and added a comment to improve readability